pypuppetdb/api/__init__.py

```diff
@@ -398,8 +404,14 @@ def nodes(self, unreported=2, with_status=False, **kwargs):
                         node['status'] = 'noop'
                     if status['failures'] > 0:
                         node['status'] = 'failed'
+                elif 'latest_report_status' in node:
+                    if node['latest_report_status'] == 'failed':
+                        node['status'] = 'failed'
                 else:
-                    node['status'] = 'unchanged'
+                    reports = [r for r in latest_reports
+                               if r.node == node['certname']]
```

Comment:
I'm confused. In line 387 I'm putting the generated reports in a list, because I needed to filter the list for the current node in the second list comprehension right here. It's not efficient, but it's just for the corner case anyway.
Making latest_reports a dict would improve readability, but I think we can't use the generator directly here.